ws-client/src/client.rs

```diff
@@ -155,9 +175,12 @@ impl<'a> WsClientBuilder<'a> {
 		self
 	}
 
-	/// Set origin header to pass during the handshake.
-	pub fn origin_header(mut self, origin: Cow<'a, str>) -> Self {
-		self.origin_header = Some(origin);
+	/// Set a custom header to pass during the handshake.
+	///
+	/// You have to ensure that headers doesn't conflict if this is called more than once.
+	/// Because it is not checked that the headers doesn't conflict.
+	pub fn custom_header(mut self, header: &'a str, value: &'a str) -> Self {
```

Comment:
I think `add_header` would be a better name. The HTTP spec doesn't define a set of headers afaik, so in a way all headers are "custom".

What do you think about changing the signature to take a `Header` directly? It'd be annoying for users that already have a `Header` to extract the name and value.

Reply:
> What do you think about changing the signature to take a Header directly? It'd be annoying for users that already have a Header to extract the name and value.

Haha, I thought the other way around that it's a pain of using `Header` directly and we would need to export it.... then I don't think it's that likely that users of the client has the `httparse crate` in their dependency tree.... but I could be wrong :)

Reply:
I was sugegsting the name + value function; yeah, I guess I also expect that whoever using this won't have a Header struct to hand; they will just know what header name+value pairs they want and this would make that easy!